components/dependency_injection/introduction.rst

```diff
@@ -104,7 +104,10 @@ like this::
         // ...
     }
 
-Then you can register this as a service as well and pass the ``mailer`` service into it::
+Then you can register this as a service as well and pass the ``mailer`` service
+into it. Instead of getting the actual ``mailer`` service, which doesn't exist
+yet when defining the ``newsletter_manager``, use the ``Reference`` class to
+get a reference to the future ``mailer`` service::
```

Comment:
This is a very big sentence. It should be split in multiple shorter ones, e.g: "When defining the `newsletter_manager` service, the `mailer` service does not exists yet. Use the `Reference` class to tell the container to inject the `mailer` service when it initializes the newsletter manager.".